### DISCUSSION TITLE
grbl mega 5x cloned axes A and B axis have rotation

### DISCUSSION THREAD

**Discussion by attk3** [winder/Universal-G-Code-Sender#2623] [Help] [answered]
Hello good day. I'm using UGS along with mega 2560, ramps 1.4, 5 drv8825, grbl mega 5x. I have a mpcnc that uses 2 motors on the X and Y axes, and I cloned the X axis in A and the Y axis in B.
The cnc works but both the cloned A axis and the B axis spin and rotate. I don't know how to solve this problem, if someone can help me it would be great since I've been stuck for several days without being able to solve it. Thanks!

I leave the result of the Grbl commands $$ , $I

$I
[VER:1.1l.20190605:]
[AXS:5:XYZXY]
[OPT:VNM,35,255,48]
ok

$$
$0 = 10 (Step pulse time, microseconds)
$1 = 255 (Step idle delay, milliseconds)
$2 = 0 (Step pulse invert, mask)
$3 = 28 (Step direction invert, mask)
$4 = 0 (Invert step enable pin, boolean)
$5 = 0 (Invert limit pins, boolean)
$6 = 0 (Invert probe pin, boolean)
$10 = 1 (Status report options, mask)
$11 = 0.020 (Junction deviation, millimeters)
$12 = 0.002 (Arc tolerance, millimeters)
$13 = 0 (Report in inches, boolean)
$20 = 0 (Soft limits enable, boolean)
$21 = 1 (Hard limits enable, boolean)
$22 = 1 (Homing cycle enable, boolean)
$23 = 27 (Homing direction invert, mask)
$24 = 200.000 (Homing locate feed rate, mm/min)
$25 = 1000.000 (Homing search seek rate, mm/min)
$26 = 200 (Homing switch debounce delay, milliseconds)
$27 = 2.000 (Homing switch pull-off distance, millimeters)
$30 = 1000 (Maximum spindle speed, RPM)
$31 = 0 (Minimum spindle speed, RPM)
$32 = 1 (Laser-mode enable, boolean)
$100 = 200.000 (X-axis travel resolution, step/mm)
$101 = 200.000 (Y-axis travel resolution, step/mm)
$102 = 800.000 (Z-axis travel resolution, step/mm)
$103 = 200.000
$104 = 200.000
$110 = 1500.000 (X-axis maximum rate, mm/min)
$111 = 1500.000 (Y-axis maximum rate, mm/min)
$112 = 300.000 (Z-axis maximum rate, mm/min)
$113 = 1500.000
$114 = 1500.000
$120 = 200.000 (X-axis acceleration, mm/sec^2)
$121 = 200.000 (Y-axis acceleration, mm/sec^2)
$122 = 200.000 (Z-axis acceleration, mm/sec^2)
$123 = 200.000
$124 = 200.000
$130 = 500.000 (X-axis maximum travel, millimeters)
$131 = 500.000 (Y-axis maximum travel, millimeters)
$132 = 120.000 (Z-axis maximum travel, millimeters)
$133 = 642.000
$134 = 642.000
ok

**Comment by attk3** [comment]
Here is a picture of the problem. Thanks.

https://ibb.co/dM6N42t

**Comment by breiler** [comment, marked as answer]
UGS expects that the status report coordinates are reported in the format XYZABC:
```
<Idle|MPos:X,Y,Z,A,B,C>
```

I suspect that grbl-5x-mega is reporting the values like this which will not work in UGS. 
```
<Idle|MPos:X1,Y1,Z,X2,Y2>
```

By enabling this option in the firmware configuration it should report cloned axes as one value:
https://github.com/fra589/grbl-Mega-5X/blob/edge/grbl/config.h#L102

**Comment by attk3** [comment]
Excellent breiler, I did what you told me and it solved the problem. Thank you very much for your help! Have a nice day!
